### PR TITLE
Fixed bug: replaced bce_loss_with_logits with bce_loss

### PR DESCRIPTION
**Pull request by amisev**
## What does this PR do?
Fixed bug in pl_examples/domain_templates/computer_vision_fine_tuning.py:
replaced binary_cross_entropy_with_logits with binary_cross_entropy.

See https://github.com/PyTorchLightning/pytorch-lightning/discussions/7081

### CODE REVIEW

**Comment by carmocca** [comment]
Why not remove the `torch.sigmoid` in forward instead?

**Comment by amisev** [comment]
Some losses in torchmetrics are required probabilities rather than scores.

File: pl_examples/domain_templates/computer_vision_fine_tuning.py [hunk 1]
```diff
@@ -213,7 +213,7 @@ def __build_model(self):
         self.fc = nn.Sequential(*_fc_layers)
 
         # 3. Loss:
-        self.loss_func = F.binary_cross_entropy_with_logits
+        self.loss_func = F.binary_cross_entropy
 
     def forward(self, x):
         """Forward pass. Returns logits."""
```
Comment (carmocca) [inline pl_examples/domain_templates/computer_vision_fine_tuning.py:219]: Can you update this docstring then?

And

```python
        # 2. Classifier (returns logits):
        x = self.fc(x)

        # 3. Convert logits to probabilities
        return torch.sigmoid(x)
```

**Comment by awaelchli** [comment]
I would go for the more numerically stable option. 
Can't we compute sigmoid outputs where we need them?

**Comment by amisev** [comment]
I removed sigmoid activation from forward pass, but anyway, even ```torchmetrics.Accuracy``` required probabilities, that's why I had to add ```torch.sigmoid``` two more times.

File: pl_examples/domain_templates/computer_vision_fine_tuning.py [hunk 2]
```diff
@@ -222,22 +222,23 @@ def forward(self, x):
...
 
     def loss(self, logits, labels):
         return self.loss_func(input=logits, target=labels)
 
     def training_step(self, batch, batch_idx):
         # 1. Forward pass:
         x, y = batch
-        y_logits = self.forward(x)
+        y_scores = self.forward(x)
+        y_logits = torch.sigmoid(y_scores)
```
Comment (awaelchli) [inline pl_examples/domain_templates/computer_vision_fine_tuning.py:237]: the variable names need to be switched now :) the sigmoid returns the probabilities and the forward returns the logits :) 
```suggestion
        y_logits = self.forward(x)
        y_scores = torch.sigmoid(y_logits)
```

File: pl_examples/domain_templates/computer_vision_fine_tuning.py [hunk 3]
```diff
@@ -222,22 +222,23 @@ def forward(self, x):
         x = self.feature_extractor(x)
         x = x.squeeze(-1).squeeze(-1)
 
-        # 2. Classifier (returns logits):
+        # 2. Classifier (returns scores):
```
Comment (awaelchli) [inline pl_examples/domain_templates/computer_vision_fine_tuning.py:225]:
```suggestion
        # 2. Classifier (returns logits):
```
this was correct :)

File: pl_examples/domain_templates/computer_vision_fine_tuning.py (hunk 2 above)
Comment (awaelchli) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py:237]: and the same in the other places :)

Comment (amisev) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py]: Done

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
great fix!

**Review by carmocca — APPROVED** [review]

**Review by s-rog — APPROVED** [review]
